src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// RuntimeEventSource is an EventSource that represents events emitted by the managed runtime.
+    /// </summary>
+    [EventSource(Name = "System.Runtime")]
+    internal sealed class RuntimeEventSource : EventSource
+    {
+        private static RuntimeEventSource s_RuntimeEventSource;
+        private EventCounter[] _counters;
+
+        private enum Counter {
+            TotalProcessTime,
+            WorkingSet,
+            HandleCount,
+            ThreadCount,
+            GCHeapSize,
+            Gen0GCCount,
+            Gen1GCCount,
+            Gen2GCCount,
+            ExceptionCount
+        }
+
+        private Timer _timer;
+
+        private const int EnabledPollingIntervalMilliseconds = 1000; // 1 second
+
+        public static void Initialize()
+        {
+            s_RuntimeEventSource = new RuntimeEventSource();
+        }
+
+        // Threads
+        // TODO
+        
+        public RuntimeEventSource(): base(EventSourceSettings.EtwSelfDescribingEventFormat)
```

Comment:
Could you please use the internal constructor with explicit Guid to avoid the expensive reflection?

(See https://github.com/dotnet/coreclr/pull/21714/ for example.)